paddle/fluid/memory/allocation/cuda_device_context_allocator.h

```diff
@@ -102,19 +109,24 @@ class CUDADeviceContextAllocator : public Allocator {
     platform::CUDADeviceGuard guard(place_.device);
     auto allocation =
         new CUDADeviceContextAllocation(memory::Alloc(place_, size));
-    // Wait for the event on stream
+// Wait for the event on stream
+#ifdef PADDLE_WITH_HIP
+    PADDLE_ENFORCE_CUDA_SUCCESS(hipEventRecord(event_, default_stream_));
```

Comment:
Better add hip support for PADDLE_ENFORCE_CUDA_SUCCESS to generate better error message.

Reply:
will address in enforce.h later as discussed in PR#30758